src/Microsoft.ServiceFabric.Services/Communication/Client/OperationRetryControl.cs

```diff
@@ -55,5 +55,11 @@ public class OperationRetryControl
         /// </summary>
         /// <value>Exception</value>
         public Exception Exception { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Func to calculate the RetryDelay for specified  retry Attempt.
+        /// </summary>
+        /// <value>Func which calculate retry delay </value>
+        public Func<int, TimeSpan> GetRetryDelay { get; set; }

```

Comment:
The retryDelay, MaxCount etc here in the retry policy is what we discussed we should fix. I think we can add a note and a bug# here. Also. instead of having the getdelay as functor, wouldnt it be simpler to have the retry policy here - for future extensibility?

Reply:
Will create  the bug to add ShouldRetry function in RetryPolicy.  RetryPolicy is not used for all the cases , for  example if user uses this constructor of [ExceptionHandlingRetryResult](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicefabric.services.communication.client.exceptionhandlingretryresult.-ctor?view=azure-dotnet#Microsoft_ServiceFabric_Services_Communication_Client_ExceptionHandlingRetryResult__ctor_System_Exception_System_Boolean_System_TimeSpan_System_Int32_)  we wont be using this retryPolicy .